tests/unit/test_utils_filesystem.py

```diff
@@ -59,3 +70,53 @@ def test_copy2_fixed_raises_appropriate_errors(create, error_type, tmpdir):
         copy2_fixed(src, dest)
 
     assert not dest.exists()
+
+
+def test_file_opener_no_file(process):
```

Comment:
These `FileOpener` tests would be better in `tests/lib/test_filesystem.py`. This file is really for tests of functionality in `pip/_internal/utils/filesystem.py`.